README.md

```diff
@@ -592,8 +595,119 @@ public Object findById(
 }
 ```
 
+Conversions
+-------------------
+
+Specification argument resolvers uses conversion mechanism to convert request string params to types of fields for which specifications have been defined.
+
+Let's consider the following code:
+  ```java
+	@Controller
+	public class SampleController {
+		
+		@Autowired
+		CustomerRepository customerRepository;
+		
+		@RequestMapping("/find")
+		public List<Customer> findByRegistrationDate(
+				@And({
+					@Spec(path = "name", params = "name", spec = Equal.class),
+					@Spec(path = "registrationDate", params = "registrationDate", spec = GreaterThanOrEqual.class)
+				}) Specification<Customer> spec) {
+			return customerRepository.findAll(spec);
+		}
+	}
+	
+	@Entity
+	public class Customer {
+		String name;
+		Date registrationDate;
+	}
+  ``` 
+When the following request will be sent to the endpoint presented above
+  ```
+  GET /find?name=John&registrationDate=2020-06-19
+  ```
+
+a `Specification<Customer>` based on fields `name` and `registrationDate` will be built.
+
+  * The type of the `name` field is a `String` type, received parameter value is always a `String` type so there is no need of conversion.  
+  * The type of the `registartionDate` field is `java.util.Date` type,  the parameter is a `String` type so `String` type will be converted into `Date` using one of available converter.
+
+##### Supported conversions  
+Specification Argument Resolver contains converters for most common java types. 
+
+List of supported conversions:
+
+  * `String -> Enum`
+  * `String -> boolean`
+  * `String -> Boolean`
+  * `String -> integer`
+  * `String -> Integer`
+  * `String -> long`
+  * `String -> Long`
+  * `String -> float`
+  * `String -> Float`
+  * `String -> double`
+  * `String -> Double`
+  * `String -> BigDecimal`
+  * `String -> Date` (default format: `yyyy-MM-dd`)
+  * `String -> LocalDate` (default format: `yyyy-MM-dd`)
+  * `String -> LocalDateTime` (default format: `yyyy-MM-dd'T'HH:mm:ss`)
+  * `String -> OffsetDateTime` (default format: `yyyy-MM-dd'T'HH:mm:ss.SSSXXX`)
+  * `String -> Instant` (default format: `yyyy-MM-dd'T'HH:mm:ss.SSSXXX`)
+  * `String -> UUID` 
+
+To use a custom format for temporal types, add `config="custom-format-value"` to `@Spec` params. 
+For example:
+```
+ @Spec(path="creationDate", spec=LessThan.class, config="dd-MM-yyyy")
+```
+
+
+In case of missing converter, [fallback mechanism](#custom-converters) will be used if one has been configured otherwise `ClassCastException` will be thrown.
+
+##### Custom converters
+The converter includes a fallback mechanism based on [Spring](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/core/convert/ConversionService.html) `ConversionService`, which is invoked when required conversion is not supported by any default converter. If the `ConversionService` supports required conversion it will be performed, if not `ClassCastException` will be thrown. 
```

Comment:
If the `ConversionService` supports required conversion it will be performed, otherwise a `ClassCastException` will be thrown.